### DISCUSSION TITLE
Object refresh on opening

### DISCUSSION THREAD

**Discussion by ichermak** [bgregs514/vscode-tm1#27] [General]
**Describe the bug**
Changes made outside of vscode are not visible when reopening an object.

**To Reproduce**
1. Open a cube rule or process in vscode-tm1
2. Add a line to the code and save it in vscode-tm1
3. Open the same rule or cube process in TM1 Architect
4. Delete the modification made in step 2 and save it with TM1 Architect
5. Reopen the cube rule or process in vscode-tm1

**Expected behavior**
Changes made outside of vscode-tm1 should be visible when reopening or refreshing and object.

**Comment by bgregs514** [comment]
Thank you for the feedback!  Although this needs to be corrected, it unfortunately is expected behavior in the current build.  Since vscode-tm1 works with local copies of the files, a routine to monitor changes __on the running instance itself__ needs to be setup to tell the extension when to refresh the local files.  This poses quite a few challenges, but is something I should be able to take a look at shortly.  An argument could also be made that we don't necessarily want the tool to automatically pull down every change that hits the system.  This may end up as a secondary "workflow" discussion.

In the meantime, I'll add a "refresh" item to the object level context menu (right click menu) so we can manually refresh.
